src/uinput.rs

```diff
@@ -175,6 +178,47 @@ impl VirtualDevice {
         self.file.write_all(bytes)
     }
 
+    /// Get the syspath representing this uinput device.
+    ///
+    /// The syspath returned is the one of the input node itself (e.g.
+    /// `/sys/devices/virtual/input/input123`), not the syspath of the device node.
+    pub fn get_syspath(&mut self) -> io::Result<PathBuf> {
+        let mut bytes = vec![0u8; 256];
+        unsafe { sys::ui_get_sysname(self.file.as_raw_fd(), &mut bytes)? };
+
+        if let Some(end) = bytes.iter().position(|c| *c == 0) {
+            bytes.truncate(end);
+        }
+
+        let s = std::str::from_utf8(&bytes).expect("invalid sys path");
```

Comment:
Mapping to `io::ErrorKind::Other` or using `from_utf8_lossy` is preferable to panicking here.

Reply:
I would normally rely on having some `Error` type, but `from_utf8_lossy` should work here since I would not expect the kernel to return something weird.